cmake/3rdPartyPackages.cmake

```diff
@@ -529,6 +529,12 @@ function(ly_force_download_package package_name)
     execute_process(COMMAND ${CMAKE_COMMAND} -E tar xf ${temp_download_target} 
          WORKING_DIRECTORY ${final_folder} COMMAND_ECHO STDOUT OUTPUT_VARIABLE unpack_result)
 
+    # For the runtime dependencies cases, we need the timestamps of the files coming from 3rdParty to be newer than the ones
+    # from the output so the new versions get copied over. The untar from the previous step preserves timestamps so they
+    # can produce binaries with older timestamps to the ones that are in the build output.
+    file(GLOB_RECURSE package_files LIST_DIRECTORIES false ${final_folder}/*)
```

Comment:
This works, but wouldn't adding --touch to the tar xf command work?  Not sure if CMAKE supports it.  Looks like it can actually take a --mtime option to set it to a specific time

Reply:
I made that attempt...., the way that CMake integrates tar makes it that only understands certain parameters, here: https://github.com/Kitware/CMake/blob/master/Source/cmcmd.cxx#L1398 and here is where it calls to do the extract: https://github.com/Kitware/CMake/blob/master/Source/cmcmd.cxx#L1457
And here is the "final" extract function, note that it doesnt pass any extra parameters: https://github.com/Kitware/CMake/blob/a6b075c3f8306a6192e66206252d76e2e5925432/Source/cmSystemTools.cxx#L1826

I was also hoping to use https://cmake.org/cmake/help/latest/command/file.html#archive-extract, but the extract function doesnt have timestamp handling.